src/oss-reproducible/Helpers/IgnoreFilter.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+using Microsoft.CST.OpenSource.Shared;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CST.OpenSource.Reproducibility
+{
+    internal class IgnoreFilter
+    {
+        private static readonly List<string> FilterText;
+
+        protected static readonly NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// Initialiizes the ignore filter using the embedded resource.
+        /// </summary>
+        static IgnoreFilter()
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            var resourceName = assembly.GetManifestResourceNames().Single(str => str.EndsWith("PackageIgnoreList.txt"));
+            using Stream? stream = assembly.GetManifestResourceStream(resourceName);
+
+            FilterText = new List<string>();
+
+            if (stream != null)
+            {
+                using StreamReader reader = new StreamReader(stream);
+                string? line;
+                while ((line = reader?.ReadLine()) != null)
+                {
+                    line = line.Trim();
+                    if (!line.Contains(":") || line.StartsWith("#") || string.IsNullOrWhiteSpace(line))
+                    {
+                        continue;  // Not a valid pattern
+                    }
+                    Logger.Trace("Adding {0} to filter.", line);
+                    FilterText.Add(line);
+                }
+            }
+            else
+            {
+                Logger.Warn("Unable to find PackageIgnoreList.txt.");
+            }
+        }
+
+        /// <summary>
+        /// Checks to see if a given file should be ignored when making a comparison.
+        /// </summary>
+        /// <param name="packageUrl"></param>
+        /// <param name="strategyName"></param>
+        /// <param name="filePath"></param>
+        /// <returns></returns>
+        internal static bool IsIgnored(PackageURL? packageUrl, string strategyName, string filePath)
+        {
+            var shouldIgnore = false;
+            filePath = filePath.Replace("\\", "/").Trim();
+            foreach (var filter in FilterText)
+            {
+                var parts = filter.Split(':', 3);
```

Comment:
Specifying 3 here seems unnecessary because you check right after that you have 3 elements.  In fact it would allow you to, for example, silently grab the first 3 elements of an "invalid" 4 element line.

Reply:
You should be able to have a 4-element line, e.g.:
```
*:*:^d:/foo/.*
```

(Though I'd expect these to be rare.)

Reply:
I don't understand how that is valid.  You only take the first three elements, so you'll be ignoring everything after the 3rd : anyway.

Reply:
Sorry, I might be misunderstanding here -- the 2nd param to .Split() is the maximum number of substrings to return, so:

```csharp
foreach (var i in "a:b:c:d:e:f:g".Split(":", 3))
 Console.WriteLine(i);
```
```
a
b
c:d:e:f:g
```